irida_import/irida_import.py

```diff
@@ -628,6 +660,97 @@ def link(self, sample_file, folder_id):
 
         return added
 
+    def upload_file_to_galaxy(self, sample_file, folder_id):
+        """
+        Upload a sample file to Galaxy
+
+        :type sample_file: SampleFile
+        :param sample_file: the sample file to upload
+        :type folder_id: ID of folder to upload file to
+        :param sample_folder_path: the folder in Galaxy to store the file in
+        :return: a list containing a single dict with the file's
+        url, id, and name.
+        """
+        self.logger.debug('Attempting to upload file')
+        added = None
+        file_path = sample_file.path
+        self.logger.debug(
+            "       Sample file's local path is" + file_path)
+        file_type = 'auto'
+        file_ext = os.path.splitext(file_path)[1]
+
+        # Assume fastq files are fastqsanger:
+        if file_ext == '.fastq':
+            file_type = 'fastqsanger'
+
+        tmp_dir = tempfile.mkdtemp()
+        tmp_file_mode = 'w+b'
+
+        try:
+            tmp_file = tempfile.NamedTemporaryFile(mode=tmp_file_mode, prefix=sample_file.name, dir=tmp_dir)
+            tmp_file.name = tmp_dir + "/" + sample_file.name
+
+            if file_ext == ".fasta":
+                headers = {'Accept': 'application/fasta'}
+            else:
+                # This accept header is used to retrieve both fastq and fast5 files.
+                # A new header type needs to be added to IRIDA to handle fast5 files
+                # separately.
+                headers = {'Accept': 'application/fastq'}
+
+            # Open the file for writing.
+            with open(tmp_file.name, tmp_file_mode) as f:
+                try:
+                    # Write the content to the file
+                    with self.irida.get(sample_file.href, headers=headers, stream=False) as resp:
+                        f.write(resp.content)
+                except:
+                    error = ("Unable to download file as it was not found:\nLocal path:{0}"
+                        ).format(sample_file.path)
+                    raise ValueError(error)
+
+            if (sample_file.upload_sha_256 == None) or (self.check_file_hash_valid(tmp_file.name, sample_file.upload_sha_256)):
+                # Copies the file into the galaxy library
+                added = self.reg_gi.libraries.upload_file_from_local_path(
```

Comment:
In this line you have `folder_id` and `file_type` explicitly stated but your first two arguments are relying on their ordered position. For readability its best to have all of them stated.

Reply:
Awesome thanks for the tips @JeffreyThiessen. Updated in 97c187bc8ab0716b21f78c299f47c20ad4f1ae00